plugins/modules/packaging/os/flatpak.py

```diff
@@ -145,6 +145,8 @@
   sample: "org.gnome.Calendar/x86_64/stable\tcurrent\norg.gnome.gitg/x86_64/stable\tcurrent\n"
 '''
 
+from distutils.version import StrictVersion
```

Comment:
Thanks for your contribution and trying to run the integration tests, @TGuimbert!
Thanks for helping people to contribute to the `flatpak` module, @felixfontein

The integration test that are in the repo right now have been disabled from the start because they depend on external resources like Flathub, which would have not have been great for CI test stability.
I did then put quite some work into converting the integration tests to rely on a local dummy repo instead and activate them in CI. I had managed to get these changes in for the `flatpak_remote` module before. I created a PR https://github.com/ansible/ansible/pull/52668 in April 2019 to achieve the same for the `flatpak` module. But I just could not seem to get the attention required to get the PR merged. My frustration about features and fixes getting merged that actually introduced regressions into the module (that would have been detected by the tests) ultimately led to me getting demotivated from continuing work on the flatpak modules. I felt the responsibility but didn't have the mandate to act on it. So I resolved that by giving up the feeling of responsibility.

So sadly, I have no idea if the integration tests have been working or not during the last year. The sane thing in the long run would be to get proper integration tests in that are part of the CI runs. If anybody would be interested, I would be happy to share the insights that I have gathered about the topic and also to get something like the old PR https://github.com/ansible/ansible/pull/52668 into working shape. But I feel that somebody else with more pull in the Ansible community would be required to drive that to get merged.

Reply:
Thanks for your feedback!
I added the changelog fragment and changed `StrictVersion` to `LooseVersion`.

For the test modification, sorry if I'm wrong, but wouldn't it be easier to integrate now that this repo is separated from the ansible one?

Reply:
@TGuimbert Good point. That is quite possible.
I just would feel a lot more motivated to get over the threshold to put in the work again if somebody would join the effort or if there was some interest from maintainers.

Reply:
@oolongbrothers I understand your point.
For what it's worth I would be happy to help!

Reply:
@TGuimbert Thank you! That's appreciated. I am working up my motivation for when I have some time on my hand. :)

Reply:
@oolongbrothers Alright, ping me when you need my input :)

Reply:
@oolongbrothers I can understand your frustration, the situation in ansible/ansible wasn't good the last years - too few people where totally overwhelmed by a huge flood of issues and PRs. The situation there is still not that much better, but at least here things are more active. @Andersson007 and me try hard to support PRs. We often cannot review all details, and have to stick to some formalities and generic knowledge (there's always a limit to our free time as well :) ), but we try to get things merged where people are actively working on. So if you can adapt the PR to this repository, I'm pretty sure we can get it merged. Assuming the tests don't fail all the time or things like that ;)

What's also possible is moving the flatpak modules into an own collection. This will be a larger effort, but would make you pretty much independent from huge repos such as this one. You can release whenever you want, merge things yourself without having to wait for a bot or one of the few humans with commit rights, and it's easier to handle issues and PRs. And you can customize the CI so it works best for these modules, like run with multiple versions of flatpak etc.. On the other hand, it means more responsibility and work. Maybe a bit much for just two modules.

In any case, I hope things are going better in this repo :) But if you are interested in creating a new collection for this (community.flatpak), it should also be possible to do that.